src/proxy/ssh/AgentProxy.ts

```diff
@@ -0,0 +1,306 @@
+  algorithm?: string;
+}
+
+/**
+ * SSH Agent Proxy
+ *
+ * Implements the SSH agent protocol over a forwarded SSH channel.
+ * This allows the Git Proxy to request signatures from the user's
+ * local ssh-agent without ever receiving the private key.
+ *
+ * The agent runs on the client's machine, and this proxy communicates
+ * with it through the SSH connection's agent forwarding channel.
+ */
+export class SSHAgentProxy extends EventEmitter {
+  private channel: Channel;
+  private pendingResponse: ((data: Buffer) => void) | null = null;
+  private buffer: Buffer = Buffer.alloc(0);
+
+  constructor(channel: Channel) {
+    super();
+    this.channel = channel;
+    this.setupChannelHandlers();
+  }
+
+  /**
+   * Set up handlers for data coming from the agent channel
+   */
+  private setupChannelHandlers(): void {
+    this.channel.on('data', (data: Buffer) => {
+      this.buffer = Buffer.concat([this.buffer, data]);
+      this.processBuffer();
+    });
+
+    this.channel.on('close', () => {
+      this.emit('close');
+    });
+
+    this.channel.on('error', (err: Error) => {
+      console.error('[AgentProxy] Channel error:', err);
+      this.emit('error', err);
+    });
+  }
+
+  /**
+   * Process accumulated buffer for complete messages
+   * Agent protocol format: [4 bytes length][message]
+   */
+  private processBuffer(): void {
+    while (this.buffer.length >= 4) {
+      const messageLength = this.buffer.readUInt32BE(0);
+
+      // Check if we have the complete message
+      if (this.buffer.length < 4 + messageLength) {
+        // Not enough data yet, wait for more
+        break;
+      }
+
+      // Extract the complete message
+      const message = this.buffer.slice(4, 4 + messageLength);
+
+      // Remove processed message from buffer
+      this.buffer = this.buffer.slice(4 + messageLength);
+
+      // Handle the message
+      this.handleMessage(message);
+    }
+  }
+
+  /**
+   * Handle a complete message from the agent
+   */
+  private handleMessage(message: Buffer): void {
+    if (message.length === 0) {
+      console.warn('[AgentProxy] Empty message from agent');
+      return;
+    }
+
+    if (this.pendingResponse) {
+      const resolver = this.pendingResponse;
+      this.pendingResponse = null;
+      resolver(message);
+    }
+  }
+
+  /**
+   * Send a message to the agent and wait for response
+   */
+  private async sendMessage(message: Buffer): Promise<Buffer> {
+    return new Promise((resolve, reject) => {
+      const length = Buffer.allocUnsafe(4);
+      length.writeUInt32BE(message.length, 0);
+      const fullMessage = Buffer.concat([length, message]);
+
+      const timeout = setTimeout(() => {
+        this.pendingResponse = null;
+        reject(new Error('Agent request timeout'));
+      }, 10000);
+
+      this.pendingResponse = (data: Buffer) => {
+        clearTimeout(timeout);
+        resolve(data);
+      };
+
+      // Send to agent
+      this.channel.write(fullMessage);
```

Comment:
Might want an extra sanity check here for the channel being open before writing 🤔